hogben/simulate.py

```diff
@@ -26,51 +51,48 @@ def simulate_magnetic(sample, angle_times, scale=1, bkg=5e-7, dq=2,
         mp (bool): whether to simulate "minus plus" spin state.
         mm (bool): whether to simulate "minus minus" spin state.
         angle_scale (float): angle to use when scaling directbeam flux.
-        directbeam_path (str): path to directbeam file to simulate with.
+        instrument (str): path to directbeam file to simulate with.
 
     Returns:
         tuple: model and simulated data for the given `sample`.
 
     """
     models, datasets = [], []
 
-    # Default path to the directbeam file for OFFSPEC when polarised.
-    if directbeam_path is None:
-        directbeam_path = importlib_resources.files('hogben.data.directbeams').joinpath('OFFSPEC_polarised_old.dat')
-        angle_scale = 0.3
-
+    instrument_DB = DB_path(instrument, polarised=True)
     # Simulate the "minus minus" spin state if requested.
     if mm:
         model, data = simulate(sample, angle_times, scale, bkg, dq,
-                               directbeam_path, angle_scale, 0)
+                               instrument_DB, angle_scale, 0)
         models.append(model)
         datasets.append(data)
 
     # Simulate the "minus plus" spin state if requested.
     if mp:
         model, data = simulate(sample, angle_times, scale, bkg, dq,
-                               directbeam_path, angle_scale, 1)
+                               instrument_DB, angle_scale, 1)
         models.append(model)
         datasets.append(data)
 
     # Simulate the "plus minus" spin state if requested.
     if pm:
         model, data = simulate(sample, angle_times, scale, bkg, dq,
-                               directbeam_path, angle_scale, 2)
+                               instrument_DB, angle_scale, 2)
         models.append(model)
         datasets.append(data)
 
     # Simulate the "plus plus" spin state if requested.
     if pp:
         model, data = simulate(sample, angle_times, scale, bkg, dq,
-                               directbeam_path, angle_scale, 3)
+                               instrument_DB, angle_scale, 3)
         models.append(model)
         datasets.append(data)
 
     return models, datasets
 
-def simulate(sample, angle_times, scale=1, bkg=5e-6, dq=2, directbeam_path=None,
-             angle_scale=0.7, spin_state=None):
+
+def simulate(sample, angle_times: np.ndarray, scale: float = 1.0, bkg: float = 5e-6, dq: float = 2,
+             instrument: str = None, angle_scale: float = 0.3, spin_state: int = None) -> tuple:
```

Comment:
`instrument` should be `Optional[str] = None`
`spin_state` should be `Optional[int] = None`